hook.c

```diff
@@ -73,6 +74,58 @@ static struct HookList Hooks = TAILQ_HEAD_INITIALIZER(Hooks);
 static struct HashTable *IdxFmtHooks = NULL;
 static HookFlags current_hook_type = MUTT_HOOK_NO_FLAGS;
 
+/**
+ * mutt_parse_charset_iconv_hook - Parse 'charset-hook' and 'iconv-hook' commands - Implements Command::parse() - @ingroup command_parse
+ */
+enum CommandResult mutt_parse_charset_iconv_hook(struct Buffer *buf, struct Buffer *s,
+                                                 intptr_t data, struct Buffer *err)
+{
+  struct Buffer *alias = mutt_buffer_pool_get();
+  struct Buffer *charset = mutt_buffer_pool_get();
+
+  int rc = MUTT_CMD_ERROR;
+  int retval = 0;
+
+  enum LookupType type = MUTT_LOOKUP_CHARSET;
+  if (data & MUTT_ICONV_HOOK)
+  {
+    type = MUTT_LOOKUP_ICONV;
+  }
+
+  retval += mutt_extract_token(alias, s, MUTT_TOKEN_NO_FLAGS);
+  retval += mutt_extract_token(charset, s, MUTT_TOKEN_NO_FLAGS);
+
+  if (retval != 0)
+  {
+    mutt_buffer_pool_release(&alias);
+    mutt_buffer_pool_release(&charset);
+    return MUTT_CMD_ERROR;
+  }
+
+  if (mutt_buffer_is_empty(alias) || mutt_buffer_is_empty(charset))
+  {
+    mutt_buffer_printf(err, _("%s: too few arguments"), buf->data);
+    rc = MUTT_CMD_WARNING;
+  }
+  else if (mutt_ch_lookup_add(type, mutt_buffer_string(alias),
+                              mutt_buffer_string(charset), err))
+  {
+    rc = MUTT_CMD_SUCCESS;
+  }
+
+  if (MoreArgs(s))
```

Comment:
We want to check for more args before we call `mutt_ch_lookup_add()`.
Currently, the hook is accepted and we get a warning.

Reply:
I called it afterwards, because I added it later - and then supected side effects if calling it before the parsing was ready.